### ISSUE TITLE
tf.keras.layers.Lambda can't infer output dimension of tf.slice

### ISSUE THREAD

**Issue by breadbread1984** [tensorflow/tensorflow#39596]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.2.0
- Python version: 3.6.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

tf.keras.layers.Lambda can't infer output dimension of tf.slice even if the input dimension is partially known.

**Describe the expected behavior**

the output dimension should be partially known as input dimension.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

```python
#!/usr/bin/python3
import tensorflow as tf;
inputs=tf.keras.Input((None,32));
results=tf.keras.layers.Lambda(lambda x: tf.slice(x, (0,0,0),(-1,tf.shape(x)[1] - 1,-1)))(inputs);
print(results.shape); # the last dimension should be 32
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by Saduf2019** [comment]
@breadbread1984 
I ran the code shared by you, please find the [gist here](https://colab.sandbox.google.com/gist/Saduf2019/efe9afccaf94dfd6e50f8d0e53e36f9d/untitled183.ipynb) and let us know id this confirms your issue

**Comment by breadbread1984** [comment]
yes it confirms tge reported issue. the last dimension should be 32 but none.

**Comment by jvishnuvardhan** [comment]
@breadbread1984 We need to use static shape (`x.shape`) instead of dynamic shape (`tf.shape(x)`) as dynamic shape doesn't support `None` in any dimension. I have updated your code to 

```
inputs=tf.keras.Input((2,32));
results=tf.keras.layers.Lambda(lambda x: tf.slice(x, (0,0,0),(-1,x.shape[1] - 1,-1)))(inputs);
print(results.shape); # the last dimension should be 32
# the above line prints (None, 1, 32)
```


Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/dd43e6edc936bad2663943147df2a247/untitled183.ipynb). Thanks!

**Comment by breadbread1984** [comment]
I met the problem when I implemented wavenet with tf2. the input tensor has input shape (batch, None, 1). Slicing on the unknown dimension (axis = 1) will cause the output dimeision become all unknown. therefore, I give the above code which can reproduce the issue.

the code you provide doesn't do slice on the known dimension.

**Comment by saikumarchalla** [comment]
Was able to  reproduce the issue using  TF 2.5 . Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/ec82c562ba62908a97387768334aa8b1/untitled80.ipynb).Thanks!

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
